engine/src/main/java/org/terasology/monitoring/impl/PerformanceMonitorImpl.java

```diff
@@ -40,8 +41,10 @@
 public class PerformanceMonitorImpl implements PerformanceMonitorInternal {
     private static final int RETAINED_CYCLES = 60;
     private static final double DECAY_RATE = 0.98;
+    private static final Activity OFF_THREAD_ACTIVITY = new NullActivity();
```

Comment:
Honestly this use of NullActivity is a little unnecessary, but - if you look at start and end activity they ignore calls from off of the main thread. We save a smidgen of performance by providing the NullActivity and not calling endActivity() over returning an Activity that does call endActivity(). The actual behavior is the same either way.